volatility/framework/plugins/windows/psscan.py

```diff
@@ -61,17 +69,88 @@ def scan_processes(cls,
             if not filter_func(mem_object):
                 yield mem_object
 
-    def _generator(self):
+    @classmethod
+    def virtual_process_from_physical(cls,
+                                      context: interfaces.context.ContextInterface,
+                                      layer_name: str,
+                                      symbol_table: str,
+                                      proc: interfaces.objects.ObjectInterface) -> \
+                Iterable[interfaces.objects.ObjectInterface]:
+        """ Returns a virtual process from a physical addressed one
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            symbol_table: The name of the table containing the kernel symbols
+            proc: the process object with phisical address
 
+        Returns:
+            A process object on virtual address space
+
+        """
+
+        # We'll use the first thread to bounce back to the virtual process
+        kvo = context.layers[layer_name].config['kernel_virtual_offset']
+        ntkrnlmp = context.module(symbol_table, layer_name = layer_name, offset = kvo)
+
+        tleoffset = ntkrnlmp.get_type("_ETHREAD").relative_child_offset("ThreadListEntry")
+        # Start out with the member offset given to us from the profile
```

Comment:
Thanks, sorry to be pain and I should've mentioned it in the original comment, but we also use layers in volatility 3, rather than spaces in volatility 2.  5:S

Reply:
oh, don't worry...just committed some additional mods...:-)